src/main/java/sparksoniq/spark/iterator/flowr/GroupByClauseSparkIterator.java

```diff
@@ -383,7 +394,44 @@ public void print(StringBuffer buffer, int indent)
             }
             buffer.append("Variable " + iterator.getVariableReference().getVariableName());
             buffer.append("\n");
-            iterator.getExpression().print(buffer, indent+1);
+            if(iterator.getExpression() != null)
+            {
+            	iterator.getExpression().print(buffer, indent+1);
+            }
+        }
+    }
+    
+    public Map<String, DynamicContext.VariableDependency> getProjection(Map<String, DynamicContext.VariableDependency> parentProjection)
+    {
+        // passing dependencies to parent
+        Map<String, DynamicContext.VariableDependency> projection = new TreeMap<String, DynamicContext.VariableDependency>();
+        projection.putAll(parentProjection);
+        for(GroupByClauseSparkIteratorExpression iterator : _expressions)
+        {
+            projection.remove(iterator.getVariableReference().getVariableName());
+        }
+        for(GroupByClauseSparkIteratorExpression iterator : _expressions)
+        {
+        	if(iterator.getExpression() == null)
+        	{
+        		String k = iterator.getVariableReference().getVariableName();
+                projection.put(k, DynamicContext.VariableDependency.FULL);
+                continue;
```

Comment:
I believe k, here represents a 'key', I would prefer the explicit name rather than a single character.